### PR TITLE
pkg/storage/etcd3: update test to time out, not panic

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Without these select statements, this test runs until the package-global
timeout and causes a panic. This change makes the test fail faster and
more legibly.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>
/kind cleanup

```release-note
NONE
```

```docs

```

/sig api-machinery
/assign @liggitt @smarterclayton @sttts @deads2k

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
/retest
flake on #108906

**Comment by stevekuznetsov** [comment]
/retest
flake on https://github.com/kubernetes/kubernetes/issues/108906

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher_test.go [hunk 1]
```diff
@@ -300,9 +300,19 @@ func TestWatchDeleteEventObjectHaveLatestRV(t *testing.T) {
 		t.Fatalf("Delete failed: %v", err)
 	}
 
-	e := <-w.ResultChan()
+	var e watch.Event
+	var wres clientv3.WatchResponse
+	select {
+	case <-time.After(wait.ForeverTestTimeout):
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher_test.go:306]: nit: would you mind switching the order of those cases?

It's more natural to first have the one that is expected to happen, and the "fallback" as a second one...

**Comment by stevekuznetsov** [comment]
@wojtek-t thanks for the review! Updated the cases :)

**Comment by stevekuznetsov** [comment]
Looks like another 1.18 data race in unit tests
/test pull-kubernetes-unit

**Comment by stevekuznetsov** [comment]
#108906 
/retest

**Comment by stevekuznetsov** [comment]
https://github.com/kubernetes/kubernetes/issues/108906
/retest

**Pull request merged**